service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ManageOrdersMidEventControllerTest.java

```diff
@@ -214,11 +252,151 @@ void reviewShouldNotAlterCaseData() {
+
+        assertThat(response.getErrors()).containsOnly("Enter a postcode for the contact");
+    }
+
+    @Test
+    void epoRemovalAddressShouldReturnErrorWhenAddressLine1IsEmpty() {
+        CaseData caseData = CaseData.builder().manageOrdersEventData(
+            buildPreventRemovalEventData(Address.builder().postcode("postcode").build())).build();
+
+        AboutToStartOrSubmitCallbackResponse response = postMidEvent(caseData, "order-details");
+
+        assertThat(response.getErrors()).containsOnly("Enter a valid address for the contact");
+    }
+
+    @Test
+    void shouldNotReturnErrorsWhenEPOOrderDetailsAreValidForRemoveToAccommodation() {
+        CaseData caseData = buildCaseData().toBuilder().manageOrdersEventData(
+            buildRemoveToAccommodationEventData(now().minusDays(4), now().plusDays(1))).build();
+
+        when(docmosisGenerationService.generateDocmosisDocument(anyMap(), eq(EPO), eq(PDF)))
+            .thenReturn(DOCMOSIS_DOCUMENT);
+
+        when(uploadService.uploadDocument(DOCUMENT_BINARIES, "Preview order.pdf", "application/pdf"))
+            .thenReturn(UPLOADED_DOCUMENT);
+
+        AboutToStartOrSubmitCallbackResponse response = postMidEvent(caseData, "order-details");
+
+        Map<String, String> mappedDocument = mapper.convertValue(DOCUMENT_REFERENCE, new TypeReference<>() {
+        });
+
+        assertThat(response.getErrors()).isEmpty();
+        assertThat(response.getData().get("orderPreview")).isEqualTo(mappedDocument);
+    }
+
+    @Test
+    void shouldNotReturnErrorsWhenEPOOrderDetailsAreValidForPreventRemoval() {
+        CaseData caseData = buildCaseData().toBuilder().manageOrdersEventData(
+            buildPreventRemovalEventData(Address.builder().addressLine1("test").postcode("SW").build()))
+            .build();
+
+        when(docmosisGenerationService.generateDocmosisDocument(anyMap(), eq(EPO), eq(PDF)))
+            .thenReturn(DOCMOSIS_DOCUMENT);
+        when(uploadService.uploadDocument(DOCUMENT_BINARIES, "Preview order.pdf", "application/pdf"))
+            .thenReturn(UPLOADED_DOCUMENT);
+
+        AboutToStartOrSubmitCallbackResponse response = postMidEvent(caseData, "order-details");
+
+        Map<String, String> mappedDocument = mapper.convertValue(DOCUMENT_REFERENCE, new TypeReference<>() {
+        });
+
+        assertThat(response.getErrors()).isEmpty();
+        assertThat(response.getData().get("orderPreview")).isEqualTo(mappedDocument);
+    }
+
     @Test
     void shouldThrowExceptionWhenMidEventUrlParameterDoesNotMatchSectionNames() {
         assertThatThrownBy(() -> postMidEvent(CaseData.builder().build(), "does-not-match"))
             .getRootCause()
             .isInstanceOf(IllegalArgumentException.class)
             .hasMessage("No enum constant uk.gov.hmcts.reform.fpl.model.order.OrderSection.DOES_NOT_MATCH");
     }
+
+    private static Map<String, String> getOrderQuestions() {
```

Comment:
Don't think this method should be needed, but if you don't want I think the name should be different and it doesn't need to be static. The name implies that it is all the order questions and then in the methods you keep adding to them.